### PR TITLE
Apply Baseline plugin to iceberg-spark

### LINKED ISSUE #161: [Baseline] Apply Baseline to iceberg-spark
Use the Baseline plugin to add linting for iceberg-spark. Part of the greater effort to apply Baseline across the project. (Issues are being created individually so that contributors can claim assignment for each of these).

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR applies Baseline to iceberg-spark and resolves #161. 

**Open Questions**

I’ve picked the default Scala formatting from Baseline. Here are a few points I want to bring attention to:
  - Baseline applies `FileLengthChecker` with 800 as `maxFileLength`. Is it ok for us?
  - Baseline applies `CyclomaticComplexityChecker` with 10 as the max value. Shall we make it 12 as in Java?
  - Baseline applies `MethodLengthChecker` with 50 as `maxLength`. Shall we make it 150 as in Java?
  - Baseline applies `ReturnChecker`. Is it ok for us?

The description of rules can be found [here](http://www.scalastyle.org/rules-dev.html).

### CODE REVIEW

**Comment by rdblue** [comment]
> I’ve changed UTF8 to Utf8 because Baseline required this.

Can we disable this rule? I don't think it is valuable and it requires a lot of code changes that result in awkward names, like Utf8Reader.

**Comment by aokolnychyi** [comment]
I tried to add `UTF8` to the list of allowed abbreviations but it didn't work for me (it worked for `SQL`, though). I will disable this rule in the next pass. Let me know what you think about the other open questions.

**Comment by aokolnychyi** [comment]
cc @rdsr @mccheah

File: spark/src/jmh/java/org/apache/iceberg/spark/source/IcebergSourceBenchmark.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ protected void appendAsFile(Dataset<Row> ds) {
         .save(table.location());
   }
 
-  protected void withSQLConf(Map<String, String> conf, Action action) {
+  protected void withSqlConf(Map<String, String> conf, Action action) {
```
Comment (rdblue) [inline spark/src/jmh/java/org/apache/iceberg/spark/source/IcebergSourceBenchmark.java]: If SQL is okay now, then let's revert these changes.

File: spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithReordering.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public Type schema(Schema schema, Supplier<Type> structResult) {
   public Type struct(Types.StructType struct, Iterable<Type> fieldResults) {
     Preconditions.checkNotNull(struct, "Cannot prune null struct. Pruning must start with a schema.");
     Preconditions.checkArgument(current instanceof StructType, "Not a struct: %s", current);
-    StructType s = (StructType) current;
+    StructType requestedStruct = (StructType) current;
```
Comment (rdblue) [inline spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithReordering.java]: I think `currenStruct` would be a better name since the variable that it is assigned from is `current`.

**Comment by rdblue** [comment]
I had some minor comments, but I don't see any bugs introduced by this. Thanks @aokolnychyi!

I think the choices you made sound good.

File: spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithReordering.java (hunk 2 above)
Comment (aokolnychyi) [reply spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithReordering.java]: I was inspired by other existing places such as [this](https://github.com/apache/incubator-iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithReordering.java#L107). It was a bit unclear what `current` means until I saw that `current = requestedType` initially. Since this class is about pruning `requested` made sense to me.

If we keep `requested` in the name, conditions are self-descriptive `requestedDecimal.scale() == decimal.scale()`.

**Comment by aokolnychyi** [comment]
I've removed `AbbreviationAsWordInName` and reverted changes required by that rule.

Also, the PR description contains a few points I want to bring attention on Scala formatting in Baseline. Let me know if we want to change defaults there.

**Comment by rdblue** [comment]
Here are some responses to the questions in the description. The rest of them sound fine to me.

> Baseline applies ReturnChecker. Is it ok for us?

Yes? What does it do?

> Baseline doesn't apply ImportOrderChecker by default. Shall we add it?

Yes, I think so.

> Baseline doesn't apply DisallowSpaceBeforeTokenChecker by default. Shall we add it?

I think so. This is consistent with the Java formatting, right?

> Baseline doesn't apply NonASCIICharacterChecker by default. Shall we add it?

Is this intended to catch fancy quotes? I think we probably don't need it. Plus, we have some areas with multi-byte characters in strings (@rdsr's recent Avro fix, for example).

**Comment by aokolnychyi** [comment]
- Enabled `ImportOrderChecker` and grouped all imports together as we do in Java. 
- Enabled `DisallowSpaceBeforeTokenChecker`.
- Let's ignore `NonASCIICharacterChecker`.

@rdblue `ReturnChecker` ensures that we don't have explicit return statements.

**Comment by rdblue** [comment]
`ReturnChecker` doesn't sound like a good idea to me. Sometimes it's easier to use an explicit return and I see no reason to disallow it.

**Comment by aokolnychyi** [comment]
+1, that's the reason why I brought up this topic. Anything else to change?

**Comment by mccheah** [comment]
Some of those are probably worth adding to Baseline proper - it seems bad to not have `ImportOrderChecker` for example. I'll file against Baseline in a bit.
